### PR TITLE
Fixed - `Interactivity API` : Some `property` access does not work well in `server directives`

### PR DESCRIPTION
**Pull request by narenin**
Trac ticket: https://core.trac.wordpress.org/ticket/61039

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/interactivity-api/class-wp-interactivity-api.php [hunk 1]
```diff
@@ -389,8 +389,10 @@ private function evaluate( $directive_value, string $default_namespace, $context
 		$path_segments = explode( '.', $path );
 		$current       = $store;
 		foreach ( $path_segments as $path_segment ) {
-			if ( isset( $current[ $path_segment ] ) ) {
+			if ( is_array( $current ) && isset( $current[ $path_segment ] ) ) {
```
Comment (sirreal) [inline src/wp-includes/interactivity-api/class-wp-interactivity-api.php]: I'm tempted to include a check like `( is_array( $current ) || $current instanceof ArrayAccess )` here 🤔

It's probably useful only in exceptional cases, but it does seem like the most correct way to check.

Do other folks have thoughts on that?

**Review by sirreal — COMMENTED** [review]
Thank you for the contribution! I think we'd need some tests for this behavior before it can be merged, is that something you could add here?

File: src/wp-includes/interactivity-api/class-wp-interactivity-api.php (hunk 1 above)
Comment (narenin) [reply src/wp-includes/interactivity-api/class-wp-interactivity-api.php]: It looks good to me, I will make these changes.

Comment (narenin) [reply src/wp-includes/interactivity-api/class-wp-interactivity-api.php]: @sirreal I have implemented the suggestions please check.

Comment (sirreal) [reply src/wp-includes/interactivity-api/class-wp-interactivity-api.php]: Thank you! My biggest concern at this point is that this behavior has no tests. Can you work on some test cases for this?



This is an example (or other files in this directory):

https://github.com/WordPress/wordpress-develop/blob/c7db7886b81da3e600dc5e6c9c961c54d1d643ef/tests/phpunit/tests/interactivity-api/wpInteractivityAPI-wp-text.php#L32-L43

@darerodz or @cbravobernal any other thoughts on what a test should look like for this or where they should go?

Comment (cbravobernal) [reply src/wp-includes/interactivity-api/class-wp-interactivity-api.php]: I added a test in https://github.com/WordPress/wordpress-develop/pull/6672
It should go in `evaluate()`  function testing.

**Comment by sirreal** [comment]
The work here was landed as part of https://core.trac.wordpress.org/changeset/58320. This PR can be closed. Thanks @narenin!

**Comment by narenin** [comment]
@sirreal Thanks for the update. I have closed the PR.

**Pull request closed without merging**